dv/cosim/spike_cosim.cc

```diff
@@ -0,0 +1,516 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "spike_cosim.h"
+#include "config.h"
+#include "decode.h"
+#include "devices.h"
+#include "log_file.h"
+#include "processor.h"
+#include "simif.h"
+
+#include <cassert>
+#include <iostream>
+#include <sstream>
+
+SpikeCosim::SpikeCosim(uint32_t start_pc, uint32_t start_mtvec,
+                       const std::string &trace_log_path, bool secure_ibex,
+                       bool icache_en)
+    : pending_iside_error(false), nmi_mode(false) {
+  FILE *log_file = nullptr;
+  if (trace_log_path.length() != 0) {
+    log = std::make_unique<log_file_t>(trace_log_path.c_str());
+    log_file = log->get();
+  }
+
+  processor =
+      std::make_unique<processor_t>("RV32IMC", "MU", DEFAULT_VARCH, this, 0,
+                                    false, log_file, nullptr, secure_ibex,
+                                    icache_en);
```

Comment:
Is there any way to pass `log_file` after constructing the object? If so, you could avoid the `unique_ptr` here, I think?